packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js

```diff
@@ -70,27 +94,29 @@ export default class QuanticColoredResultBadge extends LightningElement {
 
   /**
    * Generates the secondary color.
-   * @param {number} lightnessAmount the amount of lightness with which to update the primary color.
    * @returns {string}
    */
-  generateSecondaryColor(lightnessAmount) {
+  generateSecondaryColor() {
     const {r, g, b} = HEXToRGB(this.color);
 
     if (invalidRGBValues(r, g, b)) {
       this.invalidColor = true;
       return defaultSecondaryColor;
     }
 
-    const HSLColor = RGBToHSL(r, g, b);
-    const {h, s} = HSLColor;
-    let l = HSLColor.l;
+    const luminance =
+      (r / 255.0) * 0.2126 +
+      (g / 255.0) * 0.7152 +
+      (b / 255.0) * 0.0722
 
-    if (l >= 50) {
-      l -= lightnessAmount;
-    } else {
-      l += lightnessAmount;
-    }
+    return luminance >= 0.50 ? 'black' : 'white'
+  }
 
-    return `hsl(${h}, ${s}%, ${l.toFixed(1)}%)`;
+  renderedCallback() {
```

Comment:
Do you mind placing this back to the top? I don't know for the others but I think we mostly have the lifecycle methods at the top of the component's code.

Reply:
yeah that's what we've done everywhere else in quantic

Reply:
Ok, c'est l'habitude de React...